### PR TITLE
Issue #3222993 by acouch: Acquia deploy

### PR DESCRIPTION
**Pull request by acouch**
## Summary of changes

1. Adds Acquia support for ``fin deploy``
1. Updates ``scripts/bin/deploy.sh`` to remove patheon's git vs sftp switch (@icasimpan we could discuss but it doesn't seem necessary for each deployment)
1. Adds Acquia ``hooks`` folder for Acquia projects

## To test

- [ ] I've tested this on an internal Acquia project
- [ ] tests pass
- [ ] Read update to docs 

### Pull request author

As the author, I have verified the following (if applicable):

- [ ] Behat/Cypress test(s) are added/modified
- [ ] Unit test(s) are added/modified
- [ ] WCAG2AA test(s) are added/modified

### Pull request reviewer

As the reviewer, I have verified the following:

- [ ] I have tested the PR locally and/or in a staging environment

### CODE REVIEW

File: hooks/common/post-code-deploy/index.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (icasimpan) [inline hooks/common/post-code-deploy/index.sh:1]: How about we move all this "hooks" to `scripts/acquia`?

File: scripts/bin/deploy.sh [hunk 2]
```diff
@@ -31,18 +32,22 @@ set_perms() {
...
 }
 
 push()
 {
   add_remote
+  build
   set_perms
   git add .
   git commit -m "Build for $1"
-  [[ "$HOSTING_PLATFORM" == "pantheon" ]] && pantheon_conn_switch git  ## Must be 'git mode' in Pantheon to commit.
```
Comment (icasimpan) [inline scripts/bin/deploy.sh:43]: I need to dig deeper into this part. Looks like the "pantheon_conn_switch" was removed entirely and it's an important part when committing code to Pantheon.

Comment (icasimpan) [reply scripts/bin/deploy.sh:43]: I added back the "pantheon_conn_switch" lines in another commit.

**Comment by icasimpan** [comment]
Please check on my comments and commits @acouch. Can't test this on an Acquia environment though as I don't have access to an internal one.

As for the "pantheon_conn_switch" it's important specially that as far as I know, there's no way to disable access to pantheon dashboard web ui.

If another developer makes a shortcut to push/deploy code outside automation and flipped the switch, automation would fail.

**Comment by acouch** [comment]
@icasimpan Let's discuss the ``pantheon_conn_switch``. Why is this necessary? Once a site is setup that should not change.

File: hooks/common/post-code-deploy/index.sh (hunk 1 above)
Comment (acouch) [reply hooks/common/post-code-deploy/index.sh:1]: This sounds good. Let me test to make sure that Acquia can find that folder properly.

**Comment by acouch** [comment]
After discussion we decided to keep the git connect and remove ``[[ "$HOSTING_PLATFORM" == "pantheon" ]] && pantheon_conn_switch sftp ``

**Pull request merged**